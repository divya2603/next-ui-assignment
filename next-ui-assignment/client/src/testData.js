export const INITIAL_DATA = {
    data: [],
    pageNumber: 0
}

export const INITIAL_STATE_DATA = {
  data: {
    data: [],
    pageNumber: 0
  },
  loading : false,
};

export const API_RESPONSE = {
    "data": [{"artistId": "159260351", "artistName": "Taylor Swift"},{"artistId": "159260352", "artistName": "Drake"}],
    "pageNumber": 0
}

export const NEXT_PAGE_API_RESPONSE = {
  "data": [{"artistId": "159260351", "artistName": "Taylor Swift1"},
  {"artistId": "159260352", "artistName": "Drake1"},
  {"artistId": "159260353", "artistName": "Taylor Swift2"},
  {"artistId": "159260354", "artistName": "Drake2"}],
  "pageNumber": 0
}