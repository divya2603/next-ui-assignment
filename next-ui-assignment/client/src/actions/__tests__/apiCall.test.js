import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { 
    getData, 
    getNextPageData 
} from '../apiCall';
import { API_RESPONSE } from '../../testData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('testing async action creators', () => {
    let store;
    let fetchSearchData = API_RESPONSE
    beforeEach(() => {
        store = mockStore({});
      });
    it('creates CREATE_SUCCESS when search was successful', () => {
    //     nock('https://localhost:5000')
    //         .get('/api/search') // Route to catch and mock
    //         .reply(200, fetchSearchData); // Mock reponse code and data
  
    //   // Dispatch action to fetch to-dos
    //     return store.dispatch(getData('red'))
    //         .then(() => { // return of async actions
    //             expect(store.getActions()).toMatchSnapshot();
    //         })
    });
});