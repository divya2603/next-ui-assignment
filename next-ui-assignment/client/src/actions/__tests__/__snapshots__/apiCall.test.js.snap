// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing action creators creates CREATE_SUCCESS when creating a to-do was successful 1`] = `
Array [
  Object {
    "payload": Object {
      "createdAt": "2017-03-02T23:04:38.003Z",
      "id": 1,
      "modifiedAt": "2017-03-22T16:44:29.034Z",
      "project": "Testing",
      "title": "Example",
    },
    "type": "GET_DATA",
  },
]
`;

exports[`testing async action creators creates CREATE_SUCCESS when creating a to-do was successful 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "artistId": "159260351",
          "artistName": "Taylor Swift",
        },
        Object {
          "artistId": "159260352",
          "artistName": "Drake",
        },
      ],
      "pageNumber": 0,
    },
    "type": "GET_DATA",
  },
]
`;
